ui/app/src/models/resourceTemplate.ts

```diff
@@ -1,32 +1,64 @@
 import { ResourceType } from "./resourceType";
 
 export interface ResourceTemplate {
-    id: string,
-    name: string,
-    version: string,
-    title: string,
-    resourceType: ResourceType,
-    current: boolean,
-    properties: any,
-    system_properties: any,
-    actions: Array<TemplateAction>,
-    customActions: Array<TemplateAction>,
-    uiSchema: any
+  id: string,
+  name: string,
+  type: string,
+  description: string,
+  version: string,
+  title: string,
+  resourceType: ResourceType,
+  current: boolean,
+  properties: any,
+  allOf?: Array<any>,
+  system_properties: any,
+  actions: Array<TemplateAction>,
+  customActions: Array<TemplateAction>,
+  required: Array<string>,
+  uiSchema: any,
+  pipeline: any
 }
 
+export const sanitiseTemplateForRJSF = (template: ResourceTemplate) => {
```

Comment:
FWIW this doesn't look that horrible to me although it is obviously a hack. Perhaps worth a second pair of eyes though